src/map/entities/charentity.cpp

```diff
@@ -1509,6 +1520,11 @@ void CCharEntity::OnItemFinish(CItemState& state, action_t& action)
 
 CBattleEntity* CCharEntity::IsValidTarget(uint16 targid, uint16 validTargetFlags, std::unique_ptr<CMessageBasicPacket>& errMsg)
 {
+    // check /blockaid
+    CCharEntity* PChar = (CCharEntity*)this->GetEntity(targid, TYPE_PC);
```

Comment:
GetEntity is expensive.  There's no reason to do it since we already get an entity object from CBattleEntity::IsValidTarget - we can check blockaid after we the entity from the base call